implementations/elixir/ockam/ockam/lib/ockam/worker.ex

```diff
@@ -246,6 +255,63 @@ defmodule Ockam.Worker do
     end
   end
 
+  def schedule_idle_timeout(state) do
```

Comment:
seems simpler to return the timeout as part of the  `handle_info| handle_continue|  handle_call`  callbacks?,  and rely on genserver to track it for us.     Unless there is some functionality that we want to enforce other than plain time between received msgs

Reply:
I don't think we can do that from generic worker without intercepting all worker `handle_*` functions, which we currently don't do.
Also gen_server events are not all ockam messages and gen_server timeouts will not happen if there are gen_server events. For example our metrics poller may check secure channels using gen_server events.